server/src/main/java/org/opensearch/common/blobstore/VerifyingMultiStreamBlobContainer.java

```diff
@@ -31,4 +32,15 @@ public interface VerifyingMultiStreamBlobContainer extends BlobContainer {
      * @throws IOException if any of the input streams could not be read, or the target blob could not be written to
      */
     void asyncBlobUpload(WriteContext writeContext, ActionListener<Void> completionListener) throws IOException;
+
+    /**
+     * Creates an async callback of an {@link java.io.InputStream} for the specified blob within the container.
+     * An {@link IOException} is thrown if requesting the input stream fails.
+     * @param blobName The name of the blob to get an {@link InputStream} for.
+     * @param position The position in the blob where the next byte will be read.
+     * @param length   An indication of the number of bytes to be read.
+     * @param listener  Async listener for {@link InputStream} object which serves the input streams and other metadata for the blob
+     * @throws IOException if the input stream could not be requested or fails
+     */
+    void readBlobAsync(String blobName, long position, long length, ActionListener<InputStream> listener) throws IOException;
```

Comment:
There's nothing "verifying" or "multi stream" about this particular method, so it probably belongs in the `BlobContainer` interface with our handy new experimental tag:

```java
@ExperimentalApi
default void readBlobAsync(String blobName, long position, long length, ActionListener<InputStream> listener) throws IOException {
    throw new UnsupportedOperationException();
}
```